src/CMakeLists.txt

```diff
@@ -40,10 +42,44 @@ target_link_libraries(config
     libzstd::static
     common)
 
+
+if(LIBSESSION_UTIL_VERSIONTAG)
+    set(PROJECT_VERSION_TAG "${LIBSESSION_UTIL_VERSIONTAG}")
+    configure_file("${CMAKE_CURRENT_SOURCE_DIR}/version.c.in" "${CMAKE_CURRENT_BINARY_DIR}/version.c")
+else()
+    find_package(Git)
+    if(EXISTS "${PROJECT_SOURCE_DIR}/.git/index" AND GIT_FOUND)
+        add_custom_command(
+            OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/version.c"
+            COMMAND
+            "${CMAKE_COMMAND}"
+            "-DGIT=${GIT_EXECUTABLE}"
+            "-DPROJECT_VERSION_MAJOR=${PROJECT_VERSION_MAJOR}"
+            "-DPROJECT_VERSION_MINOR=${PROJECT_VERSION_MINOR}"
+            "-DPROJECT_VERSION_PATCH=${PROJECT_VERSION_PATCH}"
+            "-DSRC=${CMAKE_CURRENT_SOURCE_DIR}/version.c.in"
+            "-DDEST=${CMAKE_CURRENT_BINARY_DIR}/version.c"
+            "-P" "${PROJECT_SOURCE_DIR}/cmake/GenVersion.cmake"
+            DEPENDS
+            "${CMAKE_CURRENT_SOURCE_DIR}/version.c.in"
+            "${PROJECT_SOURCE_DIR}/.git/index")
+    else()
+        message(STATUS "Git was not found or this is not a git checkout. Setting version tag to 'unknown'")
+        set(PROJECT_VERSION_TAG "nogit")
+        configure_file("${CMAKE_CURRENT_SOURCE_DIR}/version.c.in" "${CMAKE_CURRENT_BINARY_DIR}/version.c")
+    endif()
+endif()
+add_library(version version.c)
```

Comment:
`version.c` to `"${CMAKE_CURRENT_BINARY_DIR}/version.c"`

may fix morgan's link error issue.

Reply:
No, that wasn't it (the `version` lib was missing from the combined archive).